nixos/modules/services/web-apps/pingvin-share.nix

```diff
@@ -0,0 +1,208 @@
+          example = "pingvin-share.domain.tdl";
+          description = mdDoc ''
+            The domain name of your instance. If null, the redirections will be made to localhost.
+          '';
+        };
+
+        https = mkOption {
+          type = types.bool;
+          default = false;
+          example = true;
+          description = mdDoc ''
+            Whether to enable HTTPS for the domain.
+          '';
+        };
+      };
+
+      backend = {
+        package = mkPackageOptionMD pkgs [ "pingvin-share" "backend" ] { };
+
+        port = mkOption {
+          type = types.port;
+          default = 8080;
+          example = 9000;
+          description = mdDoc ''
+            The port that the backend service of Pingvin Share will listen to.
+          '';
+        };
+      };
+
+      frontend = {
+        package = mkPackageOptionMD pkgs [ "pingvin-share" "frontend" ] { };
+
+        port = mkOption {
+          type = types.port;
+          default = 3000;
+          example = 8000;
+          description = mdDoc ''
+            The port that the frontend service of Pingvin Share will listen to.
+          '';
+        };
+      };
+
+      nginx = {
+        enable = mkEnableOption (mdDoc ''
+          a Nginx reverse proxy for Pingvin Share.
+        '');
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    users.groups = mkIf (cfg.group == "pingvin") {
+      pingvin = { };
+    };
+
+    users.users = mkIf (cfg.user == "pingvin") {
+      pingvin = {
+        group = cfg.group;
+        description = "Pingvin Share daemon user";
+        isSystemUser = true;
+      };
+    };
+
+    networking.firewall.allowedTCPPorts = mkIf cfg.openFirewall [ cfg.frontend.port ];
+
+    systemd.services.pingvin-share-backend = {
+      description = "Backend service of Pingvin Share, a self-hosted file sharing platform.";
+
+      wantedBy = [ "multi-user.target" "pingvin-share-frontend.service" ];
+      wants = [ "network-online.target" ];
+      before = [ "pingvin-share-frontend.service" ];
+      after = [ "network.target" ];
+
+      environment = {
+        PRISMA_SCHEMA_ENGINE_BINARY = "${pkgs.prisma-engines}/bin/schema-engine";
+        PRISMA_QUERY_ENGINE_BINARY = "${pkgs.prisma-engines}/bin/query-engine";
+        PRISMA_QUERY_ENGINE_LIBRARY = "${pkgs.prisma-engines}/lib/libquery_engine.node";
+        PRISMA_INTROSPECTION_ENGINE_BINARY = "${pkgs.prisma-engines}/bin/introspection-engine";
+        PRISMA_FMT_BINARY = "${pkgs.prisma-engines}/bin/prisma-fmt";
+
+        PORT = toString cfg.backend.port;
+        DATABASE_URL = "file:${cfg.settings.dataDir}/pingvin-share.db?connection_limit=1";
+        DATA_DIRECTORY = cfg.settings.dataDir;
+      };
+
+      path = with pkgs; [
+        cfg.backend.package
+        openssl
+        prisma-engines
+      ];
+
+      script = ''
+        cd ${cfg.backend.package}
+        prisma migrate deploy
+        prisma db seed
+        ts-node dist/src/main
```

Comment:
I think something like this should work right?
- `serviceConfig.WorkingDirectory = cfg.backend.package`
- `serviceConfig.ExecStart = "${cfg.backend.package}/bin/ts-node start"`

We don't want `bash` to be process systemd monitors.

Reply:
Indeed, I fixed that